File: python_algorithms/palindrome.py
def is_palindrome(string):
    i = 0
    j = len(string)-1
    while i < j:
        if string[i] != string[j]:
            return False
        i += 1
        j -= 1
    return True


s = input('Enter a string: ')
if is_palindrome(s):
    print(f'{s} is a palindrome.')
else:
    print(f'{s} is not a palindrome.')

File: training/java/BasicsOfJava/src/learn/Palindrome.java
package learn;

public class Palindrome {
	public static void main(String[] args) {
		int number=-49094;
		if(number<0) {
			System.out.print(number + "is it Palindrome " + isPalindrome(number*-1));
		}
		else 
		{
			System.out.print(number + " " + isPalindrome(number));
		}
	}
	
	public static boolean isPalindrome(int number) {
			int anotherNumber = 0;
			int copyNumber = number;
			while (copyNumber != 0) {
			anotherNumber = (anotherNumber * 10) + copyNumber % 10;
			copyNumber /= 10;
		}
		return anotherNumber == number;
	}
}
